app/lib/flag-route.js

```diff
@@ -1,16 +1,29 @@
-import Route from '@ember/routing/route';
+const ROUTE_LIX_ENABLED_KEY = '__route_lix_is_enabled';
 
 function generateRouteHandler(owner, key, treatmentRouteName) {
   const treatmentRoute = owner.lookup(`route:${treatmentRouteName}`);
   const flagService = owner.lookup('service:flags');
   return {
+    set(target, prop, value) {
+      const flagIsControl = !Reflect.get(target, ROUTE_LIX_ENABLED_KEY);
```

Comment:
Is there concern assuming the order of setting this flag? It seems we can just read from the service instead of assuming it is set in beforeModel.

Reply:
Yes. We need to keep the value for lix consistent until the route is re-entered. Like we can't have the effective flag change while the route is being used.

Reply:
When you say `re-entered` are you referring to rerunning model hooks, not `activate`?

Reply:
Correct. Activate happens after you've already run the model hooks.